### PR TITLE
PnP resolver

### PR DESCRIPTION
**Pull request by koshic**

### CODE REVIEW

File: src/esmockModule.js [hunk 1]
```diff
@@ -116,7 +117,8 @@ const esmockModuleId = async (parent, treeid, defs, ids, opt, mocks, id) => {
 
   if (!id) return mocks
 
-  const fileURL = opt.resolver(id, parent)
+  const resolver = await esmockResolver(opt)
```
Comment (iambumblehead) [inline src/esmockModule.js:120]: the resolver 'getter' calls could be made once inside esmockArgs.js, but either way is Okay really

**Review by iambumblehead — APPROVED** [review]
looks good

**Comment by koshic** [comment]
Next step is to create unit tests. @iambumblehead which approach do you prefer - real setup with yarn pnp or some test with mocked pnpapi module? Looks like the second one is much more simple.

**Comment by iambumblehead** [comment]
> Next step is to create unit tests. @iambumblehead which approach do you prefer - real setup with yarn pnp or some test with mocked pnpapi module? Looks like the second one is much more simple.

Whatever you think is best is okay with me and we could add or change things later any time anyway

File: src/esmockModule.js (hunk 1 above)
Comment (koshic) [reply src/esmockModule.js:120]: no problem, I'll change it.

**Review by iambumblehead — APPROVED** [review]

**Pull request merged**